src/generators/gradient_based/functions.jl

```diff
@@ -30,7 +37,8 @@ end
 """
     ∇(generator::AbstractGradientBasedGenerator, M::Models.AbstractDifferentiableModel, ce::AbstractCounterfactualExplanation)
 
-The default method to compute the gradient of the counterfactual search objective for gradient-based generators. It simply computes the weighted sum over partial derivates. It assumes that `Zygote.jl` has gradient access.
+The default method to compute the gradient of the counterfactual search objective for gradient-based generators.
+It simply computes the weighted sum over partial derivates. It assumes that `Zygote.jl` has gradient access.
 """
 function ∇(
```

Comment:
Small nitpick, but I think that it would be clearer if the hinge_loss term was added inside this method rather than inside ∂ℓ. This makes it easier to see that Probe's loss function has three terms - otherwise, the robustness term seems more like a component of the model's loss function.

Reply:
Changed it to your liking, I do agree that this is a bit cleaner.